cmd/ocm/login/cmd.go

```diff
@@ -265,9 +275,14 @@ func run(cmd *cobra.Command, argv []string) error {
 		cfg.User = ""
 		cfg.Password = ""
 	}
-	err = config.Save(cfg)
-	if err != nil {
-		return fmt.Errorf("Can't save config file: %v", err)
+
+	// If an OCM region is provided, update the config URL with the SDK generated URL
+	if cfg.RhRegion != "" {
+		regValue, err := rhRegion.GetRhRegion(context.Background(), refreshToken, args.rhRegion)
```

Comment:
so i think region discovery inside login creates a awkward scenario if you try to use `--rh-region` without an active login session. So like say i have never logged in ever. I run `ocm login --rh-region=singapore` and im met with an error "Not logged in, run the 'login' command" well thats what im trying to do...

ocm-shards.json is public, we dont actually need credentials to fetch it, but we do need a url to fetch it from (e.g. api.openshift.com/.../ocm-shards.json for prod or api.stage.openshift.com/.../ocm-shards.json for stage, etc). 

I wonder how we could make it friendlier...

Reply:
If we pass ocm login --rh-region=singapore, I get this message currently:
```
In order to log in it is mandatory to use '--token', '--user' and '--password', or '--client-id' and '--client-secret'.
You can obtain a token at: https://console.redhat.com/openshift/token .
See 'ocm login --help' for full help.
```

Also, token is mandatory right now if I want to create an SDK connection using ConnectionBuilder i.e. 

> globalConn, err := sdk.NewConnectionBuilder().Tokens(token).RhRegion("integration").BuildContext(ctx)

If I try using this command without the token, I get an error 
`either a token, an user name and password or a client identifier and secret are necessary, but none has been provided`

We can use the direct URLs (i.e. instead of sdk just call the api.openshift.com/.../ocm-shards.json directly, but since this might move to SDK, not sure if this wold be a good idea or not